python/tvm/target/target.py

```diff
@@ -113,6 +114,8 @@ def __init__(self, target, host=None):
 
     def __enter__(self):
         _ffi_api.TargetEnterScope(self)
+        if self.kind.name == "cuda":
+            self.enter_cuda_scope()
```

Comment:
It is too as-hoc to add a CUDA specific process in a general target method.

Reply:
Good suggestion, I will try to amend that.